includes/functions/updates/llms-functions-updates-600.php

```diff
@@ -0,0 +1,204 @@
+ *
+ * Queries 50 earned awards at a time and migrates their data by moving meta data
+ * to the new location and then deleting the deprecated meta values.
+ *
+ * @since [version]
+ *
+ * @param string $type Award type, either "achievement" or "certificate".
+ * @return boolean Returns `true` if there are more results and `false` if there are no further results.
+ */
+function _migrate_awards( $type ) {
+
+	$per_page = llms_update_util_get_items_per_page();
+
+	$query = new \WP_Query(
+		array(
+			'orderby'        => array( 'ID' => 'ASC' ),
+			'post_type'      => "llms_my_{$type}",
+			'posts_per_page' => $per_page,
+			'no_found_rows'  => true, // We don't care about found rows since we'll run the query as many times as needed anyway.
+			'fields'         => 'ids', // We just need the ID for the updates we'll perform.
+			'meta_query'     => array(
+				'relation' => 'OR',
+				array(
+					'key'     => "_llms_{$type}_title",
+					'compare' => 'EXISTS',
+				),
+				array(
+					'key'     => "_llms_{$type}_template",
+					'compare' => 'EXISTS',
+				),
+				array(
+					'key'     => "_llms_{$type}_image",
+					'compare' => 'EXISTS',
+				),
+			),
+		)
+	);
+
+	// Don't trigger deprecations.
+	remove_filter( 'get_post_metadata', 'llms_engagement_handle_deprecated_meta_keys', 20, 3 );
+	foreach ( $query->posts as $post_id ) {
+		_migrate_award( $post_id, $type );
+	}
+	// Re-enable deprecations.
+	add_filter( 'get_post_metadata', 'llms_engagement_handle_deprecated_meta_keys', 20, 3 );
+
+	// If there was 50 results assume there's another page and run again, otherwise we're done.
+	return ( count( $query->posts ) === $per_page );
+
+}
+
+/**
+ * Migrate meta values for a single award.
+ *
+ * Performs the following updates:
+ *   + Copies lifterlms_user_postmeta user data to the post_author property.
+ *   + Moves the title from postmeta to the post_title property.
+ *   + Moves the template relationship from meta to the post_parent property.
+ *   + Moves the award image from custom meta to the post's featured image.
+ *
+ * And then deletes the previous metadata after performing the necessary updates.
+ *
+ * @since [version]
+ *
+ * @param int    $post_id WP_Post ID.
+ * @param string $type    Award type, either "achievement" or "certificate".
+ * @return void
+ */
+function _migrate_award( $post_id, $type ) {
+
+	$obj = 'achievement' === $type ? new \LLMS_User_Achievement( $post_id ) : new \LLMS_User_Certificate( $post_id );
+
+	$updates = array(
+		'author' => $obj->get_user_id(),
+	);
+
+	$title = get_post_meta( $post_id, "_llms_{$type}_title", true );
+	if ( $title ) {
+		$updates['title'] = $title;
+	}
+
+	$template = get_post_meta( $post_id, "_llms_{$type}_template", true );
+	if ( $template ) {
+		$updates['parent'] = $template;
+	}
+	$obj->set_bulk( $updates );
+
+	_migrate_image( $post_id, $type );
+
+	delete_post_meta( $post_id, "_llms_{$type}_title" );
+	delete_post_meta( $post_id, "_llms_{$type}_template" );
+
```

Comment:
```suggestion
	if ( 'achievement' === $type ) {
		delete_post_meta( $post_id, '_llms_achievement_content' );
	}

```
per https://github.com/gocodebox/lifterlms/pull/1869#issuecomment-977730697